plugins/modules/foreman_compute_resource.py

```diff
@@ -230,16 +234,10 @@ def main():
 
     entity_dict = module.clean_params()
 
-    module.connect()
-
-    entity = module.find_resource_by_name('compute_resources', name=entity_dict['name'], failsafe=True)
-
-    entity_dict = module.handle_taxonomy_params(entity_dict)
+    if not module.desired_absent and 'provider' not in entity_dict:
```

Comment:
I think this changes the logic, as before the error was only raised when `entity=None`, so when we're creating a new one, now it's always raised when `state` is not `absent`.

Reply:
reverted to the previous logic.